pkgs/development/idris-modules/build-idris-package.nix

```diff
@@ -29,22 +34,23 @@ stdenv.mkDerivation ({
   # opts = -i ../../path/to/package
   # rather than the declarative pkgs attribute so we have to rewrite the path.
   postPatch = ''
-    sed -i *.ipkg -e "/^opts/ s|-i \\.\\./|-i ${idris-with-packages}/libs/|g"
+    sed -i ${ipkgName}.ipkg -e "/^opts/ s|-i \\.\\./|-i ${idris-with-packages}/libs/|g"
   '';
 
   buildPhase = ''
-    idris --build *.ipkg
+    idris --build ${ipkgName}.ipkg
   '';
 
   checkPhase = ''
-    if grep -q test *.ipkg; then
-      idris --testpkg *.ipkg
+    if grep -q test ${ipkgName}.ipkg; then
+      idris --testpkg ${ipkgName}.ipkg
     fi
   '';
 
   installPhase = ''
-    idris --install *.ipkg --ibcsubdir $out/libs
-    IDRIS_DOC_PATH=$out/doc idris --installdoc *.ipkg || true
+    idris --install ${ipkgName}.ipkg --ibcsubdir $out/libs
+    IDRIS_DOC_PATH=$out/doc idris --installdoc ${ipkgName}.ipkg || true
+    ${extraInstallPhase}
```

Comment:
I think using `postInstall` in the declaration of the package should work, no extra argument to the build-idris-package function needed.

Reply:
tried `postInstall` but it does not seem to do anything..

Reply:
Ah I figured it out with the help of @lheckemann: `build-idris-package` is supposed to be running these hooks. Like this:
```diff
diff --git a/pkgs/development/idris-modules/build-idris-package.nix b/pkgs/development/idris-modules/build-idris-package.nix
index 3ed1404fef7..8b298112e51 100644
--- a/pkgs/development/idris-modules/build-idris-package.nix
+++ b/pkgs/development/idris-modules/build-idris-package.nix
@@ -28,23 +28,31 @@ stdenv.mkDerivation ({
   # Some packages use the style
   # opts = -i ../../path/to/package
   # rather than the declarative pkgs attribute so we have to rewrite the path.
-  postPatch = ''
+  patchPhase = ''
+    runHook prePatch
     sed -i *.ipkg -e "/^opts/ s|-i \\.\\./|-i ${idris-with-packages}/libs/|g"
+    runHook postPatch
   '';
 
   buildPhase = ''
+    runHook preBuild
     idris --build *.ipkg
+    runHook postBuild
   '';
 
   checkPhase = ''
+    runHook preCheck
     if grep -q test *.ipkg; then
       idris --testpkg *.ipkg
     fi
+    runHook postCheck
   '';
 
   installPhase = ''
+    runHook preInstall
     idris --install *.ipkg --ibcsubdir $out/libs
     IDRIS_DOC_PATH=$out/doc idris --installdoc *.ipkg || true
+    runHook postInstall
   '';
 
 } // newAttrs)
```

Then `postInstall` will work and `extraPostInstall` isn't needed :)